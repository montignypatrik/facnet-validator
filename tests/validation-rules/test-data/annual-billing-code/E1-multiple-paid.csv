#;Facture;ID RAMQ;Date de Service;Début;Fin;Periode;Lieu de pratique;Secteur d'activité;Diagnostic;Code;Unités;Rôle;Élément de contexte;Montant Preliminaire;Montant payé;Doctor Info;DEV NOTE - À TRADUIRE: Report_DailyClaimInfo_TotalPrelAmount;DEV NOTE - À TRADUIRE: Report_DailyClaimInfo_TotalFinalAmount;DEV NOTE - À TRADUIRE: Report_DailyClaimInfo_TotalCount;Agence;Patient;Grand Total
1;128357270;15600245854;2025-01-10;08:00;;202520;55489;Aucun Secteur;;15815;0;1;;49,15;49,15;1069491-00000 | Dupont, Marie - Omnipraticien  ;98,30;98,30;2;;TEST12345678 - Test, Patient A;
2;128357315;15600247173;2025-06-15;09:00;;202520;55489;Aucun Secteur;;15815;0;1;;49,15;49,15;1069491-00000 | Dupont, Marie - Omnipraticien  ;98,30;98,30;2;;TEST12345678 - Test, Patient A;
