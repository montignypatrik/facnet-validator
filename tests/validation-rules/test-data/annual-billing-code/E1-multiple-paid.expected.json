{
  "scenarioId": "E1",
  "scenarioName": "Multiple Paid Billings (CRITICAL)",
  "description": "Code annuel facturé et payé plusieurs fois pour le même patient la même année",
  "expectedResults": [
    {
      "severity": "error",
      "category": "annual_limit",
      "message": "Code annuel 15815 facturé 2 fois et payé 2 fois pour le même patient en 2025. Maximum: 1 par an.",
      "solution": "Veuillez vérifier si les deux visites ont bien été payées. Si oui, remplacez l'une d'entre elles par une visite conforme au besoin.",
      "ruleData": {
        "monetaryImpact": 0,
        "code": "15815",
        "patient": "TEST12345678",
        "year": 2025,
        "totalCount": 2,
        "paidCount": 2,
        "unpaidCount": 0,
        "dates": ["2025-01-10", "2025-06-15"],
        "amounts": [49.15, 49.15],
        "totalPaidAmount": 98.30
      },
      "affectedRecordCount": 2
    },
    {
      "severity": "info",
      "category": "annual_limit",
      "message": "Validation complétée pour 2 enregistrement(s) de codes annuels. 1 erreur(s) détectée(s).",
      "solution": null,
      "ruleData": {
        "monetaryImpact": 0,
        "totalRecords": 2,
        "errorCount": 1,
        "totalAtRisk": 0,
        "codesChecked": ["15815"],
        "breakdown": {
          "E1": 1,
          "E2": 0,
          "E3": 0
        }
      }
    }
  ]
}
